cpp/src/arrow/sparse_tensor.h

```diff
@@ -204,6 +204,10 @@ class ARROW_EXPORT SparseTensor {
   /// \brief Return whether sparse tensors are equal
   bool Equals(const SparseTensor& other) const;
 
+  /// \brief Return dense representation of sparse tensor as tensor
+  Status ToTensor(MemoryPool* pool, std::shared_ptr<Tensor>* out) const;
+  Status ToTensor(std::shared_ptr<Tensor>* out) const;
```

Comment:
The definition of this can be written here like below.

```c++
Status ToTensor(std::shared_ptr<Tensor>* out) const {
  return ToTensor(default_memory_pool(), out);
}
```